tigers-lottery/app/src/main/java/com/example/tigers_lottery/HostedEvents/OrganizerCreateEventFragment.java

```diff
@@ -1,4 +1,142 @@
+import android.os.Bundle;
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.example.tigers_lottery.DatabaseHelper;
+import com.example.tigers_lottery.R;
+import com.example.tigers_lottery.models.Event;
+import com.google.firebase.Timestamp;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+public class OrganizerCreateEventFragment extends Fragment {
+
+    private EditText inputEventName, inputEventLocation, inputRegistrationOpens, inputRegistrationDeadline, inputEventDate, inputEventDescription, inputWaitlistLimit;
+    private CheckBox checkboxWaitlistLimit;
+    private Button btnCreateEvent;
+    private DatabaseHelper dbHelper;
+
+    public OrganizerCreateEventFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.organizer_createevent_fragment, container, false);
+
+        // Initialize views
+        inputEventName = view.findViewById(R.id.inputEventName);
+        inputEventLocation = view.findViewById(R.id.inputEventLocation);
+        inputRegistrationOpens = view.findViewById(R.id.inputRegistrationOpens);
+        inputRegistrationDeadline = view.findViewById(R.id.inputRegistrationDeadline);
+        inputEventDate = view.findViewById(R.id.inputEventDate);
+        inputEventDescription = view.findViewById(R.id.inputEventDescription);
+        checkboxWaitlistLimit = view.findViewById(R.id.checkboxWaitlistLimit);
+        inputWaitlistLimit = view.findViewById(R.id.inputWaitlistLimit);
+        btnCreateEvent = view.findViewById(R.id.btnCreateEvent);
+
+        // Initialize DatabaseHelper
+        dbHelper = new DatabaseHelper();
+
+        // Show/hide waitlist limit input based on checkbox
+        checkboxWaitlistLimit.setOnCheckedChangeListener((buttonView, isChecked) -> {
+            inputWaitlistLimit.setVisibility(isChecked ? View.VISIBLE : View.GONE);
+        });
+
+        // Set click listener for Create Event button
+        btnCreateEvent.setOnClickListener(v -> createEvent());
+
+        return view;
+    }
+
+    private void createEvent() {
+        // Capture user input
+        String eventName = inputEventName.getText().toString().trim();
+        String eventLocation = inputEventLocation.getText().toString().trim();
+        String registrationOpens = inputRegistrationOpens.getText().toString().trim();
+        String registrationDeadline = inputRegistrationDeadline.getText().toString().trim();
+        String eventDate = inputEventDate.getText().toString().trim();
```

Comment:
Add input Validation

- wait list cant open before the current date  and has to open before the wait list deadline
- wait list deadline has to be after waitlist open date and before event date
- event date should be after both wait list deadlines

waitlist opens -> waitlist closes -> event date